sqlcapture/tests/tests.go

```diff
@@ -239,3 +247,134 @@ func testCatalogPrimaryKeyOverride(ctx context.Context, t *testing.T, tb TestBac
+		},
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(1230), Exp: 0},
+			[]tuple.TupleElement{20004, []byte("123")}, // "1230e0"
+		},
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(123400), Exp: 3},
+			[]tuple.TupleElement{20009, []byte("1234")}, // "123400e3"
+		},
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(12300), Exp: -2},
+			[]tuple.TupleElement{20003, []byte("123")}, // "12300e-2"
+		},
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(12300), Exp: -5},
+			[]tuple.TupleElement{20000, []byte("123")}, // "12300e-5"
+		},
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(12300), Exp: -10},
+			[]tuple.TupleElement{19995, []byte("123")}, // "12300e-10"
+		},
+		// Zeros.
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(0), Exp: 0},
+			[]tuple.TupleElement{0, []byte(nil)}, // "0e0"
+		},
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(-0), Exp: 2},
+			[]tuple.TupleElement{0, []byte(nil)}, // "0e2"
+		},
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(0), Exp: -2},
+			[]tuple.TupleElement{0, []byte(nil)}, // "0e-2"
+		},
+		// Negatives.
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(-1), Exp: 0},
+			[]tuple.TupleElement{-20001, []byte("8:")}, // "-1e0"
+		},
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(-2), Exp: 0},
+			[]tuple.TupleElement{-20001, []byte("7:")}, // "-2e0"
+		},
+
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(-1), Exp: -1},
+			[]tuple.TupleElement{-20000, []byte("8:")}, // "-1e-1"
+		},
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(-1), Exp: -5},
+			[]tuple.TupleElement{-19996, []byte("8:")}, // "-1e-5"
+		},
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(-12309), Exp: -6},
+			[]tuple.TupleElement{-19999, []byte("87690:")}, // "-12309e-6"
+		},
+
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(-1230), Exp: -0},
+			[]tuple.TupleElement{-20004, []byte("876:")}, // "-1230e0"
+		},
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(-123400), Exp: 3},
+			[]tuple.TupleElement{-20009, []byte("8765:")}, // "-123400e3"
+		},
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(-12300), Exp: -2},
+			[]tuple.TupleElement{-20003, []byte("876:")}, // "-12300e-2"
+		},
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(-12300), Exp: -5},
+			[]tuple.TupleElement{-20000, []byte("876:")}, // "-12300e-5"
+		},
+		{
+			pgtype.Numeric{Status: pgtype.Present, Int: big.NewInt(-12300), Exp: -10},
+			[]tuple.TupleElement{-19995, []byte("876:")}, // "-12300e-10"
+		},
+	}
+	for _, test := range testCases {
+		var actual, err = sqlcapture.EncodePgNumeric(test.input)
+		require.NoError(t, err)
+		require.Equal(t, test.expected, actual)
+	}
+
+	for p1 := 0; p1 < len(testCases); p1++ {
+		for p2 := 0; p2 < len(testCases); p2++ {
```

Comment:
If you're confident that the test is properly commutative (which it should be) you can just say `p2 <= p1` here to avoid a bunch of redundant work.

Reply:
In the new test, the test cases are sorted, so this no longer applies.